### PR TITLE
added profanity filter to user sols

### PR DESCRIPTION
**Pull request by ady00**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

Added a really simple regex profanity filter - let me know if I need to add/remove words. Tackles issue #3222.

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

**Comment by bqi343** [comment]
it would be better to use a package like https://www.npmjs.com/package/bad-words 

(also, this would avoid needlessly injecting profanity into our codebase)

File: src/components/ProblemSolutions.tsx [hunk 1]
```diff
@@ -203,7 +205,9 @@ export default function ProblemSolutions({
                             : undefined
                         }
                       >
-                        {submission.solutionCode}
+                        {filter.isProfane(submission.solutionCode)
+                          ? filter.clean(submission.solutionCode)
```
Comment (bqi343) [inline src/components/ProblemSolutions.tsx:209]: why not just `filter.clean(submission.solutionCode)`?

**Review by bqi343 — CHANGES_REQUESTED** [review]
don't commit package-lock

File: src/components/ProblemSolutions.tsx (hunk 1 above)
Comment (ady00) [reply src/components/ProblemSolutions.tsx:209]: Tried filter.clean and filter.isProfane on my personal build, filter.isProfane was significantly faster on sections with more than 30-40 public user solutions.

**Comment by ady00** [comment]
final commit on #3242

**Pull request closed without merging**